src/main/java/org/jenkinsci/plugins/parallel_test_executor/TargetBranchPreviousBuildFinder.java

```diff
@@ -0,0 +1,29 @@
+package org.jenkinsci.plugins.parallel_test_executor;
+
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.mixin.ChangeRequestSCMHead;
+
+/**
+ * Returns the latest build from the target branch if the current build comes from a change request.
+ */
+@Extension(optional = true)
```

Comment:
```suggestion
@Extension
```

since we are now using the hard dep; or for that matter the whole extension point _could_ be inlined.